src/reactviews/pages/ConnectionDialog/components/trustServerCertificateDialog.component.tsx

```diff
@@ -0,0 +1,81 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { useContext } from "react";
+import { ConnectionDialogContext } from "../connectionDialogStateProvider";
+import {
+    Button,
+    Dialog,
+    DialogActions,
+    DialogBody,
+    DialogContent,
+    DialogSurface,
+    DialogTitle,
+    Link,
+    MessageBar,
+} from "@fluentui/react-components";
+
+import { locConstants } from "../../../common/locConstants";
+import { connectionCertValidationReadMoreUrl } from "../connectionConstants";
+
+export const TrustServerCertificateDialog = ({}) => {
+    const context = useContext(ConnectionDialogContext)!;
+
+    return (
+        <Dialog open={context.state.trustServerCertError !== undefined}>
+            <DialogSurface>
+                <DialogBody>
+                    <DialogTitle>
+                        {locConstants.connectionDialog.connectionErrorTitle}
+                    </DialogTitle>
+                    <DialogContent>
+                        <MessageBar
+                            intent="error"
+                            style={{ paddingRight: "12px" }}
+                        >
+                            {context.state.trustServerCertError}
+                        </MessageBar>
+                        <br />
+                        {locConstants.connectionDialog.trustServerCertMessage}
+                        <br />
+                        <br />
+                        {locConstants.connectionDialog.trustServerCertPrompt}
+                        {" " /* extra space before the 'Read More' link*/}
+                        <Link href={connectionCertValidationReadMoreUrl}>
+                            {locConstants.connectionDialog.readMore}
+                        </Link>
+                    </DialogContent>
+                    <DialogActions>
+                        <Button
+                            appearance="primary"
+                            onClick={() => {
+                                context.cancelTrustServerCertDialog();
+                                context.formAction({

```

Comment:
this should also check if it's in connectionString input mode, and add/set the property there